### PR TITLE
Implemented backend for user rank page [#135]

### PR DESCRIPTION
**Pull request by vineetkoppalkar**
[Task #135]
This pull request consists of the backend work for the tag search of user rank page. Companies have exclusive access to the user rank page, where they can see a list of all user tag scores based on a selected tag.

### CODE REVIEW

**Comment by kjanei** [comment]
Bug: When logged in as a company, the user rank page's header shows "Log in/Register" rather than "Logout"

**Comment by vineetkoppalkar** [comment]
Bug regarding the Logout button on the user rank page has been fixed by passing the user object to the user rank page such that the header can load the button properly.

File: routes/userPage.js [hunk 1]
```diff
@@ -23,6 +23,11 @@ router.patch("/:userId/clear", function (req, res, next) {
 /* GET user page, will reflect the logged-in user. */
 router.get("/", function(req, res, next){
     if (req.user) { 
+
+        console.log("~!~!~!~!~!~!~!~!~!~!~!~!");
```
Comment (kjanei) [inline routes/userPage.js]: Remove logging please

File: routes/userRankPage.js [hunk 2]
```diff
@@ -1,9 +1,70 @@
...
+const User =  require("../models/user");
+
 /* GET About Us page. */
-router.get("/", function(req, res, next) {
-    res.render("userRankPage", { user : req.user });
+router.get('/', function(req, res, next) {
+	User.find()
+        .exec()
+        .then(docs => {        	
+			res.render('userRankPage', { user: req.user, users: docs, tagName : "Please select a tag above", tagScores: "-" });
```
Comment (kjanei) [inline routes/userRankPage.js]: Excessive indentation, 4 extra spaces should suffice

File: routes/userRankPage.js [hunk 3]
```diff
@@ -1,9 +1,70 @@
...
+    var fieldTag = getFieldWithTag(tag);	//tagJava
+        	
+    User.find()
+    	.exec()
+        .then(docs => {
+        	var tagValues = [];
+
+        	var query = {};
+
+        	console.log(docs);
```
Comment (kjanei) [inline routes/userRankPage.js]: Unnecessary logging

File: routes/userRankPage.js [hunk 4]
```diff
@@ -1,9 +1,70 @@
...
+        	var tagValues = [];
+
+        	var query = {};
+
+        	console.log(docs); 	
+        	for(var i = 0; i < docs.length; ++i) {    		
+        		tagValues.push(docs[i][fieldTag]);
+        	}    		
+
+			res.render('userRankPage', { user: req.user, tagName : tag, users: docs, tagField: fieldTag });
```
Comment (kjanei) [inline routes/userRankPage.js]: Excessive indentation

File: views/question.hbs [hunk 5]
```diff
@@ -281,26 +281,26 @@
...
-                          <form action="/question/{{_id}}/reject" method="post">
-                            <input name="_method" type="hidden" value="PATCH">
-                            <button class="btn btn-dark accept-reject" type="submit">Reject</button>
-                          </form>
-                        </div>
-                      </div>
-                        {{/unlessEq}}
-                    {{/eq}}
-                  {{/unless}}
+                    {{#unless @root.question.hasAccepted}}
```
Comment (kjanei) [inline views/question.hbs]: Why were 2 spaces added to each line here? The code within `{{unless}}` seems to be n the same level as the previous `div`.

**Pull request merged**